src/components/src/main/java/org/apache/jmeter/config/CSVDataSet.java

```diff
@@ -120,31 +121,39 @@ private Object readResolve(){
      */
     @Override
     public void setProperty(JMeterProperty property) {
-        if (property instanceof StringProperty) {
-            final String propName = property.getName();
-            if ("shareMode".equals(propName)) { // The original name of the property
-                final String propValue = property.getStringValue();
-                if (propValue.contains(" ")){ // variables are unlikely to contain spaces, so most likely a translation
-                    try {
-                        final BeanInfo beanInfo = Introspector.getBeanInfo(this.getClass());
-                        final ResourceBundle rb = (ResourceBundle) beanInfo.getBeanDescriptor().getValue(GenericTestBeanCustomizer.RESOURCE_BUNDLE);
-                        for(String resKey : CSVDataSetBeanInfo.getShareTags()) {
-                            if (propValue.equals(rb.getString(resKey))) {
-                                if (log.isDebugEnabled()) {
-                                    log.debug("Converted {}={} to {} using Locale: {}", propName, propValue, resKey, rb.getLocale());
-                                }
-                                ((StringProperty) property).setValue(resKey); // reset the value
-                                super.setProperty(property);
-                                return;
-                            }
-                        }
-                        // This could perhaps be a variable name
-                        log.warn("Could not translate {}={} using Locale: {}", propName, propValue, rb.getLocale());
-                    } catch (IntrospectionException e) {
-                        log.error("Could not find BeanInfo; cannot translate shareMode entries", e);
-                    }
-                }
+        if (!(property instanceof StringProperty)) {
+            super.setProperty(property);
+            return;
+        }
+        final String propName = property.getName();
+        if (!"shareMode".equals(propName)) {
+            super.setProperty(property);
+            return;
+        }
+        final String propValue = property.getStringValue();
+        // variables are unlikely to contain spaces, so most likely a translation
+        if (!propValue.contains(" ")) {
+            super.setProperty(property);
+            return;
+        }
+        try {
+            ResourceBundle rb = (ResourceBundle) Introspector
+                    .getBeanInfo(this.getClass())
+                    .getBeanDescriptor()
+                    .getValue(GenericTestBeanCustomizer.RESOURCE_BUNDLE);
+
+            Optional<String> first = Arrays.stream(CSVDataSetBeanInfo.getShareTags())
+                    .filter(resKey -> propValue.equals(rb.getString(resKey)))
```

Comment:
Would it be even more readable when we extract `rb.getString(resKey)` into a  `map` step before filtering?
That would be
```
.map(rb::getString)
.filter(propValue::equals)
```

Reply:
I think that would then change the behaviour.
If we mapped to `rb::getString` we would then return that value rather than the `resKey`.
Or have I missed something?

Reply:
I agree with @vlsi, although it could be more kindly said.

Reply:
I think the return was only there to stop the `log.warn` from being called, regardless I've reverted the controversial parts of this change :)